### PR TITLE
fix(Scripts/Underbog: Underbat's Tentacle Lash should be casted only …

### LINKED ISSUE #13972: [The Underbog] NPC - Sporebat - Tentacle Lash
### Current Behaviour

Casts "Tentacle Lash" on current target. Does not damage targets behind him.

### Expected Blizzlike Behaviour

Similar to "Tail Lash" and should attack targets behind it

https://www.wowhead.com/tbc/spell=34171
>Extra damage to an enemy when behind them

https://www.wowhead.com/tbc/guide/underbog-dungeon-strategy-burning-crusade-classic
>Underbats are the one mob that melee dps should always attack from the front. This ability will cause a lot of unnecessary damage to you that could otherwise be completed removed and spare your healer the trouble.

https://www.icy-veins.com/tbc-classic/the-underbog-dungeon-guide
>Tentacle Lash
This is one of the few mobs you want to attack from the front regardless of role to avoid taking large spikes of damage.

### Source

https://www.wowhead.com/tbc/spell=34171
https://www.wowhead.com/tbc/guide/underbog-dungeon-strategy-burning-crusade-classic
https://www.icy-veins.com/tbc-classic/the-underbog-dungeon-guide

### Steps to reproduce the problem

.go cr 138310
Take a second character with you and try to attack him from behind

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/55957fbb6e24f2fb6df57c92b829ad42498afee0

### Operating system

Windows 10 x64

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by UltraNix**
…on targets behind the bat.

Fixes #13972

## Issues Addressed:

- Closes #13972

## Tests Performed:

- Not tested.

## How to Test the Changes:

`.go cr 138310`
Take a second character with you and try to attack him from behind

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1673176236867407100.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+--
+UPDATE `creature_template` SET `AiName`='', `ScriptName`='npc_underbat' WHERE `entry`=17724;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=17724 AND `source_type`=0;
+
+DELETE FROM `spelldifficulty_dbc` WHERE `ID`=34171;
+INSERT INTO `spelldifficulty_dbc` VALUES
+(34171,34171,37956,0,0);
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1673176236867407100.sql:7]: why are we changing dbc values?

File: src/server/scripts/Outland/CoilfangReservoir/underbog/underbog.cpp [hunk 2]
```diff
@@ -40,7 +42,52 @@ class spell_fungal_decay : public AuraScript
...
+        _scheduler.Schedule(2200ms, 6900ms, [this](TaskContext context)
+        {
+            if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, [&](Unit* u)
+            {
+                return u->IsAlive() && !u->IsPet() && me->IsWithinCombatRange(u, 20.f) && !me->HasInArc(M_PI, u);
+            }))
+            {
+                DoCast(target, SPELL_TENTACLE_LASH);
+            }
+            context.Repeat(5700ms, 9700ms);
```
Comment (acidmanifesto) [inline src/server/scripts/Outland/CoilfangReservoir/underbog/underbog.cpp:70]: where are these repeat values cominf from?

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1673176236867407100.sql (hunk 1 above)
Comment (UltraNix) [reply data/sql/updates/pending_db_world/rev_1673176236867407100.sql:7]: We are not changing dbc values - we are adding missing entries.

File: src/server/scripts/Outland/CoilfangReservoir/underbog/underbog.cpp (hunk 2 above)
Comment (UltraNix) [reply src/server/scripts/Outland/CoilfangReservoir/underbog/underbog.cpp:70]: From SAI script.

File: data/sql/updates/pending_db_world/rev_1673176236867407100.sql (hunk 1 above)
Comment (acidmanifesto) [reply data/sql/updates/pending_db_world/rev_1673176236867407100.sql:7]: I am researching and checking dbcs from 243 to 10.1 and i do not see 34171 ever exsisting in the spell difficulties dbc/db2 files ever.

**Comment by acidmanifesto** [comment]
Marking as need feedback.
Reference my comment.
https://github.com/azerothcore/azerothcore-wotlk/pull/14566#discussion_r1089725191

File: data/sql/updates/pending_db_world/rev_1673176236867407100.sql (hunk 1 above)
Comment (UltraNix) [reply data/sql/updates/pending_db_world/rev_1673176236867407100.sql:7]: That's why I'm adding.

**Comment by Gultask** [comment]
I can go in there and check in Retail

**Comment by acidmanifesto** [comment]
> I can go in there and check in Retail

Im not certain how checking retail is going to show that as being/absent in the dbc table. The spell was never in the dbc/db2 of any expansion since the spells introduction in tbc.

**Comment by Gultask** [comment]
> > I can go in there and check in Retail
> 
> Im not certain how checking retail is going to show that as being included/absent in the dbc table. The spell was never in the dbc/db2 of any expansion since the spells introduction in tbc.

Just to check if a different spell is cast. I'm not well-versed in dbc to know what it changes 😛

**Comment by UltraNix** [comment]
> > I can go in there and check in Retail
> 
> Im not certain how checking retail is going to show that as being included/absent in the dbc table. The spell was never in the dbc/db2 of any expansion since the spells introduction in tbc.

As you can see, it's in DB and not in DBC files. That tables has been created to add more entries to it, so we can emulate the missing spells casted on different difficulties instead of hardcodes in c++ script.

**Comment by Nyeriah** [comment]
spelldifficulty_dbc complements the dbc file, it's incomplete and may at times contain incorrect data, as has happened before with some Ulduar spells. It doesn't reflect the dbc file and as such I don't see an issue with this approach

**Comment by Gultask** [comment]
There's 300 rows in spelldifficulty_dbc, too. I doubt one more will be a problem.

**Comment by Maelthyrr** [comment]
Hi, sorry for the delay. I confirm that works fine. Tentacle Lash is cast on the player behind him. Just need to fix the merge conflict.

**Comment by UltraNix** [comment]
Fixed.

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**